modules/ossm-deploying-automatic-gateway-injection.adoc

```diff
@@ -12,70 +12,193 @@ When deploying a gateway, you must opt-in to injection by adding an injection la
-    - port: 443
-      name: https
-  ---
-  apiVersion: apps/v1
-  kind: Deployment
-  metadata:
-    name: istio-ingressgateway
-    namespace: istio-ingress
-  spec:
-    selector:
-      matchLabels:
-        istio: ingressgateway
-    template:
-      metadata:
-        annotations:
-          inject.istio.io/templates: gateway <1>
-        labels:
-          istio: ingressgateway <2>
-          sidecar.istio.io/inject: "true" <3>
-      spec:
-        containers:
-        - name: istio-proxy
-          image: auto <4>
-  ---
-  apiVersion: rbac.authorization.k8s.io/v1 <5>
-  kind: Role
-  metadata:
-    name: istio-ingressgateway-sds
-    namespace: istio-ingress
-  rules:
-    - apiGroups: [""]
-      resources: ["secrets"]
-      verbs: ["get", "watch", "list"]
-  ---
-  apiVersion: rbac.authorization.k8s.io/v1
-  kind: RoleBinding
-  metadata:
-   name: istio-ingressgateway-sds
-   namespace: istio-ingress
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: Role
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway
+      labels:
+        istio: <ingressgateway>
+        sidecar.istio.io/inject: "true"
+    spec:
+      containers:
+      - name: istio-proxy
+        image: auto
+----
+
+. Enable gateway injection by setting the `sidecar.istio.io/inject` field to `"true"`. When connecting to a revisioned control plane, replace with `istio.io/rev: revision-name`.
++
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istio-ingressgateway
+  namespace: istio-ingress
+spec:
+selector:
+  matchLabels:
+    istio: ingressgateway
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway
+      labels:
+        istio: <ingressgateway>
+        sidecar.istio.io/inject: "true"
+    spec:
+      containers:
+      - name: istio-proxy
+        image: auto
+----
+
+. Set the `image` field to `auto` so that the image automatically updates each time the pod starts.
++
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istio-ingressgateway
+  namespace: istio-ingress
+spec:
+  selector:
+    matchLabels:
+      istio: ingressgateway
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway
+      labels:
+        istio: <ingressgateway>
+        sidecar.istio.io/inject: "true"
+    spec:
+      containers:
+      - name: istio-proxy
+        image: auto
+----
+
+. Set up roles to allow reading credentials for TLS.
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
   name: istio-ingressgateway-sds
+  namespace: istio-ingress
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "watch", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+ name: istio-ingressgateway-sds
+ namespace: istio-ingress
+roleRef:
+apiGroup: rbac.authorization.k8s.io
+kind: Role
+name: istio-ingressgateway-sds
```

Comment:
Please remove 2 leading spaces in lines 132-134.

Reply:
@rh-tokeefe I did a mistake here, I'm sorry. Lines 132-134 should start with 2 spaces.